lib/intercom/request.rb

```diff
@@ -110,7 +110,7 @@ def parse_body(decoded_body, response)
       rescue JSON::ParserError => _
         raise_errors_on_failure(response)
       end
-      raise_errors_on_failure(response) if parsed_body.nil?
+      raise_errors_on_failure(response, default: true) if parsed_body.nil?
```

Comment:
I'm not sure I love the pattern of passing in this default.

Is there any reason for not having the UnexpectedError be the general base case? (i.e. an `else`)

Reply:
Agreed. It felt not ideal, but enough to try to start a conversation. Thanks for the quick response!

When I added a default `else`, a number of tests failed. It looks like the gem is dependent on this method not having a fallback in this method in [here](https://github.com/intercom/intercom-ruby/blob/master/lib/intercom/request.rb#L78).

[Line 111](https://github.com/intercom/intercom-ruby/blob/master/lib/intercom/request.rb#L111) could see a similar problem, though, where a non 4/500 throws a JSON parse error but method doesn't raise and we keep moving down through the logic.

A deeper fix would be nice, but it felt like might end up potentially breaking behavior some users are relying on and pushing the version to 4.0. If you're up for that I'd be happy to spend some time working on a fix like that. Also happy to tweak this a different way to keep it from being a breaking change if there are other ideas to try.

Reply:
Hmm interesting. 🤔 Being honest I'm not entirely certain of why that code relies so heavily on there not being a default error here, it feels like a perfectly reasonable scenario. 

If you're willing to look for a deeper fix that would be great! Ideally I'd like to not use this flag behaviour behaviour which is somewhat complicated as it is. But if you don't have the time to do that we could look into it here.

Bumping to a version 4.0 isn't really a problem.